src/pages/patientView/mutationalSignatures/MutationalSignatureBarChart.tsx

```diff
@@ -0,0 +1,346 @@
+        const lengthLegendObjects = getLengthLabelEntries(legendEntries);
+        const legendInfoBoxes = formatLegendObjectsForRectangles(
+            lengthLegendObjects,
+            legendEntries,
+            labels,
+            this.props.version
+        );
+        const legendLabelBoxes: JSX.Element[] = [];
+        legendInfoBoxes.forEach((item: DrawRectInfo) => {
+            legendLabelBoxes.push(
+                <rect
+                    x={xScale(item.start)}
+                    y={15}
+                    fill={item.color}
+                    width={
+                        xScale(item.end)! - xScale(item.start)! > 0
+                            ? xScale(item.end)! - xScale(item.start)!
+                            : 6
+                    }
+                    height="15px"
+                />
+            );
+        });
+        return legendLabelBoxes;
+    }
+
+    @computed get getSubLabelsLegend() {
+        const groupedData = _.groupBy(
+            getColorsForSignatures(this.props.data),
+            'group'
+        );
+        const labels = Object.keys(groupedData);
+        const legendLabels = getColorsForSignatures(this.props.data).map(
+            entry => ({
+                group: entry.group,
+                label: entry.label,
+                color: entry.colorValue,
+                subcategory: entry.subcategory,
+                value: entry.label,
+            })
+        );
+        const uniqueSubLabels = legendLabels.filter(
+            (value, index, self) =>
+                index ===
+                self.findIndex(
+                    t =>
+                        t.group === value.group &&
+                        t.subcategory === value.subcategory
+                )
+        );
+
+        const centerOfBoxes = this.formatColorBoxLegend;
+        const subLabelsForBoxes = formatLegendObjectsForRectangles(
+            [uniqueSubLabels.length],
+            uniqueSubLabels,
+            uniqueSubLabels.map(item => item.subcategory!),
+            this.props.version
+        );
+        const legendLabelsChart: JSX.Element[] = [];
+        subLabelsForBoxes.forEach((item: LabelInfo, i: number) => {
+            legendLabelsChart.push(
+                <VictoryLabel
+                    x={
+                        centerOfBoxes[i].props.x +
+                        0.5 * centerOfBoxes[i].props.width
+                    }
+                    y={37}
+                    width={this.props.width}
+                    text={item.category}
+                    style={{ fontSize: '10px' }}
+                    textAnchor={'middle'}
+                />
+            );
+        });
+        return legendLabelsChart;
+    }
+
+    @action getLabelsForTooltip(data: IMutationalCounts[]): string[] {
+        return getColorsForSignatures(data).map(item => item.label);
+    }
+
+    public render() {
+        return (
+            <div id={'mutationalBarChart'} style={{ paddingTop: '10px' }}>
```

Comment:
we should never use id attribute.  what's the purpose here?

Reply:
Removed `id` and created a ref for the dom element.